accounts-db/src/tiered_storage/readable.rs

```diff
@@ -55,7 +55,15 @@ impl<'accounts_file, M: TieredAccountMeta> ReadableAccount
 {
     /// Returns the balance of the lamports of this account.
     fn lamports(&self) -> u64 {
-        self.meta.lamports()
+        if let Some(lamports) = self.meta.lamports_from_meta() {
+            return lamports;
+        }
+
+        // If the meta does not have the lamports, then it must inside the optional
+        // fields.
+        self.meta
+            .lamports_from_optional_fields(self.account_block)
+            .unwrap()
```

Comment:
And, here's the updated lamports() function.  Note that as we currently have 30 bits, this PR currently doesn't use the concept of storing the minimum non-zero lamports in the footer.  It will be in a separate PR.